### PR TITLE
Creates a default experiment at API server set up time

### PR DESCRIPTION
**Pull request by rileyjbauer**
This experiment is named `Default` and is used whenever a `CreateRun` request is received without specifying a containing experiment.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1089)

### CODE REVIEW

File: backend/test/initialization/initialization_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+type InitializationTest struct {
+	suite.Suite
+	namespace        string
+	experimentClient *api_server.ExperimentClient
+}
+
+// Check the namespace have ML job installed and ready
+func (s *InitializationTest) SetupTest() {
+	if !*runIntegrationTests {
```
Comment (Ark-kun) [inline backend/test/initialization/initialization_test.go]: What's the purpose of this switch?

BTW, `go test` supports test case regex which allows only running the tests you want.

Comment (rileyjbauer) [reply backend/test/initialization/initialization_test.go]: This prevents the test from running as part of the the travis CI tests

File: backend/src/apiserver/resource/resource_manager.go [hunk 2]
```diff
@@ -214,6 +217,31 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 		return nil, util.NewInternalServerError(err, "Failed to create a workflow for (%s)", workflow.Name)
 	}
 
+	hasExperiment := false
```
Comment (IronPan) [inline backend/src/apiserver/resource/resource_manager.go]: could you move the logic here into a util function, maybe named something like setExperimentIfNotPresent?

File: backend/src/apiserver/storage/default_experiment_store.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	defaultExperimentDBValue = sq.Eq{"DefaultExperimentId": ""}
+)
+
+type DefaultExperimentStoreInterface interface {
+	GetDefaultExperimentId() (string, error)
+	SetDefaultExperimentId(id string) error
+}
+
+// Implementation of a DefaultExperimentStoreInterface. This stores the default experiment's ID,
+// which is created as the API server is initialized.
```
Comment (IronPan) [inline backend/src/apiserver/storage/default_experiment_store.go]: the first time the API server is initialized.

File: backend/src/apiserver/storage/default_experiment_store.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+type DefaultExperimentStore struct {
+	db *DB
+}
+
+func (s *DefaultExperimentStore) InitializeDefaultExperimentTable() error {
```
Comment (IronPan) [inline backend/src/apiserver/storage/default_experiment_store.go]: this can be a private method.

File: test/initialization-test/OWNERS [hunk 5]
```diff
@@ -0,0 +1,6 @@
+approvers:
```
Comment (IronPan) [inline test/initialization-test/OWNERS]: add your name?

File: backend/src/apiserver/resource/resource_manager.go (hunk 2 above)
Comment (rileyjbauer) [reply backend/src/apiserver/resource/resource_manager.go]: Done

**Pull request merged**